### ISSUE TITLE
deb package for debian/ubuntu

### ISSUE THREAD

**Issue by holms** [enzo1982/freac#8]
Any plans on that? Seems to be pretty basic need.

**Comment by enzo1982** [comment]
I plan to provide packages once fre:ac 1.1 goes beta.

However, if you or someone else would like to create package scripts now, feel free to do so. I'll gladly accept them for the project.

**Comment by holms** [comment]
I'd be glad to but I have issues actually compiling it. Anyway I'll come back in here then later on, as I'm interested in this app :)

**Comment by enzo1982** [comment]
OK, maybe I help you to compile fre:ac? What are the issues?

**Comment by forrcaho** [comment]
I'm going to try to work on this. I don't want to stop anyone else from taking it up, since I have a steep learning curve and can't promise I'll have it done anytime soon, but if someone else _does_ take it up, please keep me in the loop.

**Comment by geoiii68** [comment]
I think this is a worth while project and a good learning experience for me.
I have nothing better to so here I go!
I have no idea what I am doing but google is my friend.

**Comment by N0rbert** [comment]
Some people [still expect](https://askubuntu.com/q/1302195/66509) that the Fre:ac sources have `debian` folder to compile the application for Debian-like systems. But what is interesting - this application [was never released](https://repology.org/project/freac/versions) for such systems. 

So having deb package support is still actual.

**Comment by enzo1982** [comment]
@geoiii68 That would be fantastic! Thank you for taking on this project!

Unfortunately, I didn't have time to create Debian packaging recipes myself yet, so your help would be greatly appreciated. Note that fre:ac depends on enzo1982/smooth and enzo1982/BoCA, so those should probably be packaged first.

If you have any questions, feel free to ask and I will try to help. Cannot help with the actual packaging process though - you'll probably need to ask in some Debian forum for that.

**Comment by geoiii68** [comment]
I am glad you guys are watching this because I am stuck. I have done smooth and boca maybe not correctly but I'm not getting any errors with it. I have done everything I know to do, I am ready to package but it's not working. I will keep searching and trying.
Details to come.

**Comment by geoiii68** [comment]
@N0rbert 
Can you elaborate?
Do you mean debian folder is not needed and do something different?
Or, do I need to create it and its contents?
I am creating it right now.

**Comment by geoiii68** [comment]
Never mind I saw you comment at ask ubuntu

**Comment by enzo1982** [comment]
Yes, I guess there needs to be a separate folder for all the files related to .deb packaging. Ideally, this would be packaging/debian, but if the Debian packaging tools expect it to be a debian folder at top level, that's ok as well.

If you already have some recipes/scripts for packaging smooth and BoCA, you can share them here. I'll have some time tomorrow and could have a look at them to see if the packaging is ok.

**Comment by DavidPesticcio** [comment]
To assist the people here that have some spare time to work on this issue, or to perhaps encourage some collaboration amongst those that do so that it gets some positive momentum collectively, the links below may be of value in understanding how to build the dependencies of fre:ac in order to produce suitable packages that can be made available to distributions that need them...

**How it's done for Arch based distributions:** (works fine for me on Manjaro!)
https://aur.archlinux.org/packages/freac

**How packages can be created with FPM for a variety of distributions:**
https://fpm.readthedocs.io/en/v1.13.1/intro.html

HTH! :smile: :+1:

**Comment by Robert-Ernst** [comment]
I am working right now on it.
See here:  [Debian Bug Tracker](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1034440)

Credits to David Vantyghem (@InfoLibre) for creating the original Request for Packaging on Sat, 15 Apr 2023 on the Debian Bug Tracker. This is also how I got my attention to this.

@enzo1982 If you don't mind I would like to ask you some technical questions so the packaging can go smoother. How is your availability and what is a good way to communicate with you?

**Comment by enzo1982** [comment]
@Robert-Ernst Thank you for stepping up! It would be great to have Debian packages for fre:ac.

Please contact me via email. You'll find my personal email in the program, readme and the source files. I'll try to respond within a day. If it helps speed things up, we can also arrange a call (that would probably work best for me on a weekend).

**Comment by Robert-Ernst** [comment]
Hello @enzo1982,
I've sent you a lengthy email on 27th April 2023 with further steps on how to proceed.
Greets Ernst

**Comment by enzo1982** [comment]
> Hello @enzo1982, I've sent you a lengthy email on 27th April 2023 with further steps on how to proceed. Greets Ernst

Thanks, just sent you a response. Sorry for the delay.

**Comment by InfoLibre** [comment]
When a debian package will be available, will you add it in each release?
